homeassistant/components/zha/config_flow.py

```diff
@@ -53,48 +224,306 @@ async def async_step_user(self, user_input=None):
         ]
 
         if not list_of_ports:
-            return await self.async_step_pick_radio()
+            return await self.async_step_manual_pick_radio_type()
 
         list_of_ports.append(CONF_MANUAL_PATH)
 
         if user_input is not None:
             user_selection = user_input[CONF_DEVICE_PATH]
+
             if user_selection == CONF_MANUAL_PATH:
-                return await self.async_step_pick_radio()
+                return await self.async_step_manual_pick_radio_type()
 
             port = ports[list_of_ports.index(user_selection)]
-            dev_path = await self.hass.async_add_executor_job(
-                usb.get_serial_by_id, port.device
+            self._device_path = port.device
+
+            if not await self._detect_radio_type():
+                # Did not autodetect anything, proceed to manual selection
+                return await self.async_step_manual_pick_radio_type()
+
+            self._title = (
+                f"{port.description}, s/n: {port.serial_number or 'n/a'}"
+                f" - {port.manufacturer}"
+                if port.manufacturer
```

Comment:
We try to avoid multi-line ternary operator expressions.

Reply:
We're going to fix this in a next PR with a bigger overhaul of the titles.